### DISCUSSION TITLE
Remote Connection issue

### DISCUSSION THREAD

**Discussion by terrorindeed** [supersaiyanmode/PyWebOSTV#82] [General]
Hello, I followed the setup, and got pywebostv running on my OSMC Vero 4k+ box. I'm building a python script or a few, to control my Oppo 203 via the Vero 4k+ .Part of that process is auto switching hdmi sources when triggered.

I'm still in the testing phase with the connection script.

I got the remote connection notification once, pressed yes. But obviousily didn't have the store object correct. Now it seems I cannot reinitialize the remote connection prompt on my LG C1.

Any ideas?
Also, I'm very new to python. I've learned a lot in 25 days utilizing chatgpt, but feel I'm at a stage where I may need further assistance.

**Comment by terrorindeed** [comment]
I've corrected the syntax error. I can now re establish the remote connection. I will attempt to store, and get back.

**Reply by terrorindeed** [reply]
Store with client_key and access token have been stored.

**Reply by supersaiyanmode** [reply]
Looks like you've been able to solve the issue, am I right? If you have any other questions -- please use one thread/discussion (this one; please post a reply here) instead of reviving an older discussion.

From the other thread:

> If I can start again,
When I initilize source_control.set_source(sources[2]), the tv does switch hdmi inputs correctly, but the time delay is roughly 15 seconds. On the otherhand, my script to initiate the http get request to the oppo is essentially instanteous. Without writing a delay in my oppo playback script, is there anything i can do to increase the speediness of the webostv command?

Please post the source code -- without it there's no way for any of us to help you. I have no idea what "Vero 4k+" is -- but I am assuming its WebOS on a non-LG TV? I have so many questions: How are you sure this is a library problem -- Is there a huge delay from when you call the API to the point when the [payload goes out via a WebSocket connection](https://github.com/supersaiyanmode/PyWebOSTV/blob/master/pywebostv/connection.py#L196)? Have you tried it with an LG TV? Do you still see a huge latency? What are the specs of Vero? Is it possible your network is slow? Is it possible that "Vero" is slow?

**Reply by terrorindeed** [reply]
Thank you!

vero 4k+ is a linux box running Kodi,  I'm using it as a GUI frontend, and Oppo 203 as the playback device. 
At the moment, I am only using the below script as I am still testing. 
In a seperate script, the http get request from the Vero 4k+ to the Oppo 203 is instanteous, no delay. 
I am using a 55' LC C1 webos 6.3.2-423. This is where I am seeing the latency, none from the vero 4k+ to the oppo. 
Vero 4k+ is fairly quick and specs are:
Quad Core 1.6Ghz 64-bit processor
2GB DDR3 RAM and 16GB eMMC storage [1]
Built in 802.11 ac/n/g/b WiFi with Bluetooth 4.0
IR and RF receiver
2x USB 2.0 ports
micro SD card slot
3.5mm analogue audio output
Optical (SPDIF) audio output
HDMI 2.0 with CEC (control your Vero 4K + with your TV remote)
Dynamic Refresh Rate adjustment (with 23.976/24p support)
Gigabit Ethernet

The LG C1 is on wifi, the other devices are hard wired. I can try the LG on ethernet and test.

```
import os
import json
from pywebostv.discovery import *
from pywebostv.connection import *
from pywebostv.controls import *

# Define the file path where the store object will be saved
store_file_path = os.path.expanduser("~/.kodi/store.txt")

# Check if the store file already exists
if os.path.exists(store_file_path):
    # If the store file exists, load the store object from the file
    with open(store_file_path, "r") as store_file:
        store = json.load(store_file)
else:
    # If the store file doesn't exist, create an empty store object
    store = {}

client = WebOSClient('192.168.1.135')
client.connect()
for status in client.register(store):
    if status == WebOSClient.PROMPTED:
        print("Please accept the connect on the TV!")
    elif status == WebOSClient.REGISTERED:
        print("Registration successful!")

source_control = SourceControl(client)
sources = source_control.list_sources()    # Returns a list of InputSource instances.
source_control.set_source(sources[2])      # .set_source(..) accepts an InputSource instance.
```

**Reply by terrorindeed** [reply]
sorry, I'm not sure why my code is bolded and larger text file, thats not how it shows in my .py

**Reply by supersaiyanmode** [reply]
You may need to write `store` back to `store_file_path` after registration is successful.

Also, did you try getting the latency until the [payload goes out via a WebSocket connection](https://github.com/supersaiyanmode/PyWebOSTV/blob/master/pywebostv/connection.py#L196)? Maybe try using the `timeit` module.

**Comment by terrorindeed** [comment]
No difference using wired connection to LG C1 (which is FE ~100mbps).

Correction, the time to switch hdmi inoputs is roughly 7 seconds, not 15.
Tested both on wifi and wired on the LG C1.

**Comment by terrorindeed** [comment]
I changed nothing at all, but restarted the Vero 4K+ Linux Kodi box, and now the command is also instanteous, it’s actually a littler faster than my other script. I’m calling hdmi_switching.py in my playback.py script. Now it seems I may need to add a slight delay 250ms before calling the wrapped script. Thanks. I think I will be good from here. Next I will be implementing another http get request to register ‘stop’ where I will employ Pywebostv again and switch back to the vero 4K+ device. Sent from my iPhoneOn Jan 24, 2023, at 9:16 PM, Srivatsan Iyer ***@***.***> wrote:﻿
You may need to write store back to store_file_path after registration is successful.
Also, did you try getting the latency until the payload goes out via a WebSocket connection? Maybe try using the timeit module.

**Comment by terrorindeed** [comment]
I'm having a minor issue, which i have isolated to my hdmi_switching.py

After the script runs successfully, the vero 4k+ is unresponsive for 2-3 minutes.
So if I select a film for playback, and script switches to hdmi 2, if i were to change my mind and decide to go back to the vero 4k+ and kodi, the GUI is unresponsive, i can howver still access the system via SSH.

Does my script need to 'client.disconnect()' at the end?

**Comment by terrorindeed** [comment]
i have fixed the unresponsness in my previous code. The script switches to hdmi [2] quickly and efficiently. 
I am now looking to switch back to hdmi [1] when certain conditions are met, i have the first figured out which is an http get request from the oppo 203, but i need another condition satisified to make this work properly.

What do i do to get the current hdmi source?

ive tried 
'# Connect to the LG webOS TV
    client = WebOSClient('192.168.1.135')
    client.connect()
    app = ApplicationControl(client)
    apps = app.list_apps()                            # Returns a list of `Application` instances.

    # Get the current source information
    app_id = app.get_current()                        # Returns the application ID (string) of the foreground app.
    current_app = [x for x in apps if app_id == x["id"]][0]
    source_info = current_app["title"]
'

as im looking to save the webos output to ~./kodi/tv_info.txt for later calling. 

thoughts? thank you
